Source/DafnyLanguageServer/Workspace/TextDocumentLoader.cs

```diff
@@ -182,15 +196,35 @@ bool loadCanceled
           errorReporter.ReportBoogieError(outcomeError);
         }
 
-        var diagnostics = errorReporter.GetDiagnostics(document.Uri).OrderBy(d => d.Range.Start).ToList();
-        concurrentDictionary.AddOrUpdate(methodPosition, diagnostics, (_, _) => diagnostics);
+        return errorReporter.GetDiagnostics(document.Uri).OrderBy(d => d.Range.Start).ToList();
+      });
+
+      var result = implementationTasks.Select(it => it.ObservableStatus.Select(async _ => {
+        if (it.CurrentStatus is VerificationStatus.Completed) {
+          var itDiagnostics = await diagnostics[it];
+          var methodPosition = it.Implementation.tok.GetLspPosition();
+
+          concurrentDictionary.AddOrUpdate(methodPosition, itDiagnostics, (_, _) => itDiagnostics);
+        }
+
+        if (it.CurrentStatus is VerificationStatus.Running) {
+          // For backwards compatibility
+          notificationPublisher.SendStatusNotification(document.TextDocumentItem, CompilationStatus.VerificationStarted, it.Implementation.Name);
+        }
 
         return document with {
+          VerificationTasks = implementationTasks,
           VerificationDiagnosticsPerMethod = concurrentDictionary.ToImmutableDictionary(),
           CounterExamples = counterExamples.ToArray(),
         };
-      }).ToList();
-      return documentTasks.Select(documentTask => documentTask.ToObservable()).Merge();
+      }).SelectMany(t => t.ToObservable())).Merge().Replay();
```

Comment:
Can you please give a meaningful name to `t`?

Reply:
Also, please consider putting the part before `      }).SelectMany(t => ` in a separate method.
What does `Replay()` does here? I'm not sure I understand.